documentation: more detailed description re FollowFD/UnfollowFD (#2228)
cilium/tetragon: docs/content/en/docs/concepts/tracing-policy/selectors.md

### FollowFD action

The `FollowFD` action allows to create a mapping using a BPF map between file
descriptors and filenames. It however needs to maintain a state
correctly, see [`UnfollowFD`](#unfollowfd-action) and
[`CopyFD`](#copyfd-action) related actions.

`FollowFD` is typically used at hook points where a file descriptor and its
associated file name appear together. The kernel function `fd_install`

> is a case in point.

Author: No, I just thought the mentioning of a processes file descriptor table might cause confusion opposite the eBPF table where Tetragon keeps associations between file descriptors and file names. I'll put the earlier sentence back in.

the index of the file descriptor argument and the index of the name argument in
the call.

While the mapping between the file descriptor and file name remains in place
(i.e. between `FollowFD` and `UnfollowFD` for the same file descriptor) tracing
policies may refer to file names where file descriptors

> would otherwise have to serve instead. This offers greater convenience and

Author: Yes, I have no issue with that.

allows more functionality to reside inside the kernel, thereby reducing
overhead.

For instance, assume that you want to prevent writes into file
`/etc/passwd`. The system call `sys_write` only receives a file descriptor

> (not a file name) as argument. Yet with a bracketing pair for `FollowFD`

Author: Ok

> (not a file name) as argument. Yet with a bracketing pair for `FollowFD`

Author: I am using a metaphor that compares `FollowFD` and `UnfollowFD` to an opening and closing bracket, respectively. The reference to a file name appears in between, like an element between two brackets in mathematical set notation; hence the pair consisting of `FollowFD` and `UnfollowFD` is bracketing the reference to the file name, so to speak. Is it clearer now?

the index of the file descriptor argument and the index of the name argument in
the call.

While the mapping between the file descriptor and file name remains in place

> (i.e. between `FollowFD` and `UnfollowFD` for the same file descriptor) tracing

Reviewer: ```suggestion
(that is, between `FollowFD` and `UnfollowFD` for the same file descriptor) tracing
```

note: we generally try to avoid use of latin abbreviations. See: https://docs.cilium.io/en/stable/contributing/docs/docsstyle/#don-t-use-latin-abbreviations.
